fiftyone/plugins/definitions.py

```diff
@@ -139,6 +140,20 @@ def js_bundle_server_path(self):
         if self.has_js:
             return os.path.join(self.server_path, self.js_bundle)
 
+    @property
+    def js_bundle_hash(self):
+        """A hash of the plugin's JS bundle file."""
+        if not self.has_js:
+            return None
+
+        try:
+            with open(self.js_bundle_path, "rb") as f:
+                h = hashlib.sha1()
+                h.update(f.read())
+                return h.hexdigest()
+        except:
+            return None
+
```

Comment:
This `except` clause is intentionally broad because we're treating this as a best-effort operation. There are not any specific errors that we expect to recover from. In the worst case, this will simply revert back to existing behavior rather than e.g. failing to load the plugin at all.